src/SSA_stepper.jl

```diff
@@ -128,15 +128,15 @@ function DiffEqBase.__init(jump_prob::JumpProblem,
      sizehint!(u,numsteps_hint)
      sizehint!(t,numsteps_hint)
    else
-     sizehint!(u,2)
-     sizehint!(t,2)
+     sizehint!(u,save_start+save_end)
+     sizehint!(t,save_start+save_end)
    end
 
     integrator = SSAIntegrator(prob.f,copy(prob.u0),prob.tspan[1],prob.tspan[1],prob.p,
                                sol,1,prob.tspan[1],
                                cb,_saveat,save_everystep,save_end,cur_saveat,
                                opts,tstops,1,false,true)
-    cb.initialize(cb,u[1],prob.tspan[1],integrator)
+    cb.initialize(cb,copy(prob.u0),prob.tspan[1],integrator)
```

Comment:
Do we even need the copy here or can just pass `prob.u0`? @ChrisRackauckas is there anything in the wrapping `DiscreteCallback` that would require a copy?

Reply:
Yeah, so this should be calling into
https://github.com/SciML/DiffEqJump.jl/blob/784cddb360167f6b9b808af444fb80422b8023da/src/aggregators/ssajump.jl#L48

There `initialize!` shouldn't modify `u`, so I think it should be safe to pass `prob.u0` directly.

Reply:
it can use `integrator.u` which already does the copy.